### PR TITLE
fixes meat slabs from butchered animals not having edible component

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
fix: you can eat meat from butchered mobs
/:cl:
FAQ
why old  way no work? i dont know
does this works? yes tested with cat, monkey, polar bear from icebox and another cat

### CODE REVIEW

**Comment by Qustinnus** [comment]
go figure out why it broke in the first place before you break more shit by not doing so.

**Comment by cacogen** [comment]
I'm curious how you knew to try this to fix it

**Comment by LemonInTheDark** [comment]
I'd like to know why it was broken too, if only so I know what to look for in future

**Comment by Tlaltecuhtli** [comment]
@LemonInTheDark 
https://github.com/tgstation/tgstation/blob/596f61c35707796995b396d867ac66772dc862ba/code/datums/components/butchering.dm#L120-L125

this loop causes bug,  removing it fixes it but i dont know it does

**Comment by Tlaltecuhtli** [comment]
which triggers this https://github.com/tgstation/tgstation/blob/c69eb59325c570a59470217a0428235473a1658e/code/game/atoms.dm#L1635
that removes edible component

File: code/datums/components/butchering.dm [hunk 1]
```diff
@@ -114,15 +114,15 @@
 		var/obj/sinew = V
 		var/amount = meat.guaranteed_butcher_results[sinew]
 		for(var/i in 1 to amount)
-			results += new sinew (T)
+			results +=  new sinew (T)
 		meat.guaranteed_butcher_results.Remove(sinew)
 
-	for(var/obj/item/carrion in results)
+	for(var/obj/item/food/carrion in results)
```
Comment (LemonInTheDark) [inline code/datums/components/butchering.dm:120]: This'll break things that get passed in which have the meat mat, but aren't meat.
Do an `if(istype(carrion, /obj/item/food))` check instead

Comment (LemonInTheDark) [reply code/datums/components/butchering.dm:120]: I will webedit this I swear to god

Comment (LemonInTheDark) [reply code/datums/components/butchering.dm:120]: TLALLLLLLLLL

**Pull request closed without merging**